src/main/java/com/epam/aidial/core/limiter/RateLimiter.java

```diff
@@ -96,71 +94,125 @@ public Future<LimitStats> getLimitStats(String deploymentName, ProxyContext cont
         dayTokenStats.setTotal(limit.getDay());
         limitStats.setDayTokenStats(dayTokenStats);
-        TokenLimitStats minuteTokenStats = new TokenLimitStats();
+
+        ItemLimitStats minuteTokenStats = new ItemLimitStats();
         minuteTokenStats.setTotal(limit.getMinute());
         limitStats.setMinuteTokenStats(minuteTokenStats);
+
+        ItemLimitStats hourRequestStats = new ItemLimitStats();
+        hourRequestStats.setTotal(limit.getRequestHour());
+        limitStats.setHourRequestStats(hourRequestStats);
+
+        ItemLimitStats dayRequestStats = new ItemLimitStats();
+        dayRequestStats.setTotal(limit.getRequestDay());
+        limitStats.setDayRequestStats(dayRequestStats);
+
         return limitStats;
     }
 
-    private ResourceDescription getResourceDescription(String deploymentName, ProxyContext context) {
-        String path = getPath(deploymentName);
+    private ResourceDescription getResourceDescription(ProxyContext context, String path) {
         String bucketLocation = BlobStorageUtil.buildUserBucket(context);
         return ResourceDescription.fromEncoded(ResourceType.LIMIT, bucketLocation, bucketLocation, path);
     }
 
-    private RateLimitResult checkLimit(ResourceDescription resourceDescription, Limit limit) {
+    private RateLimitResult checkLimit(ProxyContext context, Limit limit) {
+        long timestamp = System.currentTimeMillis();
+        RateLimitResult tokenResult = checkTokenLimit(context, limit, timestamp);
+        if (tokenResult.status() != HttpStatus.OK) {
+            return tokenResult;
+        }
+        return checkRequestLimit(context, limit, timestamp);
+    }
+
+    private RateLimitResult checkTokenLimit(ProxyContext context, Limit limit, long timestamp) {
+        String tokensPath = getPathToTokens(context.getDeployment().getName());
+        ResourceDescription resourceDescription = getResourceDescription(context, tokensPath);
         String prevValue = resourceService.getResource(resourceDescription);
-        RateLimit rateLimit = ProxyUtil.convertToObject(prevValue, RateLimit.class);
+        TokenRateLimit rateLimit = ProxyUtil.convertToObject(prevValue, TokenRateLimit.class);
         if (rateLimit == null) {
             return RateLimitResult.SUCCESS;
         }
-        long timestamp = System.currentTimeMillis();
         return rateLimit.update(timestamp, limit);
     }
 
-    private Void updateLimit(ResourceDescription resourceDescription, long totalUsedTokens) {
-        resourceService.computeResource(resourceDescription, json -> updateLimit(json, totalUsedTokens));
+    private RateLimitResult checkRequestLimit(ProxyContext context, Limit limit, long timestamp) {
+        String tokensPath = getPathToRequests(context.getDeployment().getName());
+        ResourceDescription resourceDescription = getResourceDescription(context, tokensPath);
+        RateLimitResult[] result = new RateLimitResult[1];
```

Comment:
you can make this class mutable to avoid passing array in lambda

Reply:
added a comment to explain the reason for doing so